MapboxNavigation/NavigationViewController.swift

```diff
@@ -20,7 +20,7 @@ public protocol NavigationViewControllerDelegate {
     /**
      Called when the user arrives at the destination.
      */
-    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt destination: MGLAnnotation)
+    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt destination: RouteStep)
```

Comment:
RouteStep isn’t a destination. Arriving at a step means something different than arriving at a destination. I agree that MGLAnnotation is a poor fit for this method, but then we should pass in a Waypoint, not a RouteStep.